### PR TITLE
Use ies

### PR DESCRIPTION
**Pull request by joakim-hove**
**Approach**
On top of: https://github.com/equinor/ert/pull/2412

With this PR the `ies_enkf_initX()` implementation is used also for the `std_enkf` module. 

## Pre review checklist

- [x] Added appropriate labels

### CODE REVIEW

**Comment by joakim-hove** [comment]
This is ready for review.

With this PR the std_enkf module uses the same internal implementation of analysis update as the ies module - i.e. although there are still superficially two modules remaining, there is only one underlying implementation.

File: libres/lib/include/ert/analysis/ies/ies.hpp [hunk 1]
```diff
@@ -39,8 +39,7 @@ void init_update(void *arg, const bool_vector_type *ens_mask,
                  const matrix_type *R, const matrix_type *dObs,
                  const matrix_type *E, const matrix_type *D, rng_type *rng);
 
-void initX(const std::variant<double, int> &truncation,
-           config::inversion_type ies_inversion, const matrix_type *Y0,
+void initX(const config::config_type *ies_config, const matrix_type *S,
```
Comment (dafeda) [inline libres/lib/include/ert/analysis/ies/ies.hpp:42]: What do you think of adding a docstring to this in Doxygen format, something like:

```cpp
/**
 * @brief 
 * 
 * @param ies_config 
 * @param S 
 * @param R 
 * @param E 
 * @param D 
 * @param X 
 */
```

Comment (joakim-hove) [reply libres/lib/include/ert/analysis/ies/ies.hpp:42]: Personally I think Doxygen means "old and irrelevant" - so I would prefer not to.

Comment (dafeda) [reply libres/lib/include/ert/analysis/ies/ies.hpp:42]: What format of docstrings do you prefer?

Comment (joakim-hove) [reply libres/lib/include/ert/analysis/ies/ies.hpp:42]: My general attitude is that I am skeptical to documentation strings, because I feel they very often are stale and irrelevant[1]. The formatted strings (typically doxygen) even more so, because I suspect the guy inserting that comment used a shortcut in his editor with thinking much about it - i.e. it screams: "Comment without love".

My preferred way to handle comments is that when something is *really* complex/broken/.... and I see no obvious way to improve the problem away I write a comment. The comment can be quite extensive, it is in free form and typically for developers actually working with the problematic code - i.e. in the .cpp file. In addition to the text in the comment the mere fact that I have actually added a comment should signal "here be dragons" for the pour soul coming after me.

But I am pushing 50 and have no spine left - will do as encouraged.


[1]: Due to `pydoc` I do feel quite different about this in Python - here the comments/document strings are available to a much larger audience and it feels meaningful to write them.

Comment (dafeda) [reply libres/lib/include/ert/analysis/ies/ies.hpp:42]: > My general attitude is that I am skeptical to documentation strings, because I feel they very often are stale and irrelevant[1]. The formatted strings (typically doxygen) even more so, because I suspect the guy inserting that comment used a shortcut in his editor with thinking much about it - i.e. it screams: "Comment without love".

The staleness is in my opinion due to docstrings containing too much detail.
In this specific case I would write that `initX` is equivalent to a standard EnKF implementation but that it uses an iterative ensemble smoother under the hood or something like that.

> My preferred way to handle comments is that when something is _really_ complex/broken/.... and I see no obvious way to improve the problem away I write a comment. The comment can be quite extensive, it is in free form and typically for developers actually working with the problematic code - i.e. in the .cpp file. In addition to the text in the comment the mere fact that I have actually added a comment should signal "here be dragons" for the pour soul coming after me.

I think this is fine, but should ideally be dealt with fast.
In other words, such comments should not grow old, which they often do.

> But I am pushing 50 and have no spine left - will do as encouraged.

The spine hardens with age!

**Comment by joakim-hove** [comment]
>In this specific case I would write that initX is equivalent to a standard EnKF implementation but that it uses an iterative ensemble smoother under the hood or something like that.

Well - that is actually wrong. It uses the standard EnKF implementation which is an internal part of the iterated algorithm. But that aside: I think your comment illustrates an interesting point, when writing comments we feel inclined to document the journey we have completed - that is not really interesting to someone coming after us.

**Comment by dafeda** [comment]
> > In this specific case I would write that initX is equivalent to a standard EnKF implementation but that it uses an iterative ensemble smoother under the hood or something like that.
> 
> Well - that is actually wrong. 
It uses the standard EnKF implementation which is an internal part of the iterated algorithm. But that aside: I think your comment illustrates an interesting point, when writing comments we feel inclined to document the journey we have completed - that is not really interesting to someone coming after us.

I thought that the iterated algorithm was for the case when the posterior is not Gaussian and when predictions and state vectors were non-linearly related.
That is, I thought the iterated algorithm made fewer assumptions than EnKF.
In any case, the journey completed provides context which is of great importance to those coming after us IMO.

**Comment by joakim-hove** [comment]
> I thought that the iterated algorithm was for the case when the posterior is not Gaussian and when predictions and state vectors were non-linearly related.
That is, I thought the iterated algorithm made fewer assumptions than EnKF.

Both the EnKF and iterated are for the nonlinear case, however the iterated makes smaller steps - i.e. less severe effects of nonlinearity.

> In any case, the journey completed provides context which is of great importance to those coming after us IMO.

Here we disagree - or at least I think this will lead to very much comments.

File: libres/lib/include/ert/analysis/ies/ies.hpp (hunk 1 above)
Comment (joakim-hove) [reply libres/lib/include/ert/analysis/ies/ies.hpp:47]: I have added a comment which I am happy with.

**Comment by joakim-hove** [comment]
Have updated the comment in the .hpp slightly and added a more verbose comment in the .cpp file

**Review by dafeda — APPROVED** [review]
LGTM 👍
